### PR TITLE
FIX improve legacy_expression_tool.md documentation

### PR DESCRIPTION
**Pull request by fgalan**
Continuation of PR #1405

### CODE REVIEW

File: scripts/legacy_expression_tool/README.md [hunk 1]
```diff
@@ -225,8 +234,10 @@ All                                                    1        1   2
 #### Execution with `expressionlanguage` set to `jexlall`
 
 When executing the script with `expressionlanguage` set to `jexlall`, the script will look for all the documents
-containing legacy expressions or the parameter `expressionLanguage`. This would change the number of documents found,
-and the statistics will be different. Running the script with the option set to `jexl` would not have impact on  the statistics. 
+containing legacy expressions (in some of the expression capable fields) or the existence of the `expressionLanguage` field.
```
Comment (fgalan) [inline scripts/legacy_expression_tool/README.md:237]: It would be great if the list of "expression capable fields" could be added here.

File: scripts/legacy_expression_tool/README.md [hunk 2]
```diff
@@ -225,8 +234,10 @@ All                                                    1        1   2
...
+This would change the number of documents found, and the statistics will include extra documents.
+
+Running the script with the option set to `jexl` would not include such extra documents in statistics.
```
Comment (fgalan) [inline scripts/legacy_expression_tool/README.md:240]: "statistics will be different" or "impact of statistics" doesn't describe the exact differences or impact.

My suggested wording try to be more precise, although I don't know if I'm fully right.

Please improve if needed.

File: scripts/legacy_expression_tool/README.md [hunk 3]
```diff
@@ -235,7 +246,4 @@ change the`expressionLanguage` field in the document. This means that the legacy
 `expressionLanguage` field will still be set to the default value or legacy. This would make expression evaluation to
 fail, propagating the value of the attribute as the expression literal to the context broker.
 
-In order to avoid this, it is recommended to set `expressionlanguage` to `jexl`, so the script will set the
-`expressionLanguage` field to `jexl` in the documents where the legacy expressions are replaced. This will ensure that  
-modified documents include the `expressionLanguage` field set to `jexl` and the expressions are evaluated correctly.
-This is specially important when replacing expressions at device level.
+In sum, it is recommented use always the `--expressionlangauge` parameter (note that is't optional) and doing it with a value different from `ignore`.
```
Comment (fgalan) [inline scripts/legacy_expression_tool/README.md:249]: I have tried to simplify this, replacing a paragraph a bit hard to follow (at least for me :) by a simple statement saying it's a bad idead to use `ignore`.

File: scripts/legacy_expression_tool/README.md (hunk 2 above)
Comment (mapedraza) [reply scripts/legacy_expression_tool/README.md:240]: You are right. I agree the rewording describes better that particular scenario.

File: scripts/legacy_expression_tool/README.md (hunk 3 above)
Comment (mapedraza) [inline scripts/legacy_expression_tool/README.md:249]: It is better to suggest `jexl` value for `--expresionlanguage`. As an example, using `delete` option would not help.

```suggestion
To avoid this, it is recommented use always the `--expressionlangauge` parameter set to `jexl` (note that is't optional) and doing it with a value different from `ignore`.
```

Comment (mapedraza) [reply scripts/legacy_expression_tool/README.md:249]: It is better to suggest `jexl` value for `--expresionlanguage`. As an example, using `delete` option would not help.

```suggestion
To avoid this, it is recommented use always the `--expressionlangauge` parameter set to `jexl` (note that is't optional) and doing it with a value different from `ignore`.
```

File: scripts/legacy_expression_tool/README.md [hunk 4]
```diff
@@ -235,7 +246,4 @@ change the`expressionLanguage` field in the document. This means that the legacy
 `expressionLanguage` field will still be set to the default value or legacy. This would make expression evaluation to
 fail, propagating the value of the attribute as the expression literal to the context broker.
 
-In order to avoid this, it is recommended to set `expressionlanguage` to `jexl`, so the script will set the
-`expressionLanguage` field to `jexl` in the documents where the legacy expressions are replaced. This will ensure that  
-modified documents include the `expressionLanguage` field set to `jexl` and the expressions are evaluated correctly.
-This is specially important when replacing expressions at device level.
+To avoid this, it is recommented use always the `--expressionlangauge` parameter set to `jexl` (note that is't optional) and doing it with a value different from `ignore`.
```
Comment (fgalan) [inline scripts/legacy_expression_tool/README.md:249]:
```suggestion
To avoid this, it is recommended use always the `--expressionlangauge` parameter set to `jexl`, so doing it a value different from `ignore` will be used.
```

File: scripts/legacy_expression_tool/README.md (hunk 1 above)
Comment (mapedraza) [reply scripts/legacy_expression_tool/README.md:237]: The script looks for expressions in the following fields:

- active.expression
- active.entity_name
- active.reverse
- attributes.expression
- attributes.entity_name
- attributes.expression
- commands.expression
- endpoint
- entityNameExp
- explicitAttrs

File: scripts/legacy_expression_tool/README.md [hunk 5]
```diff
@@ -104,6 +104,15 @@ The list of possible arguments that the scripts accepts are:
...
+* `jexl`: changes expressions from legacy to JEXL equivalence in the [fields where expressions may be used](#fields-in-which-expressions-may-be-used) (based on the translation dictionary specified by `--translations`). In addition, it set `expressionLanguage` field to `jexl` (no matter if the field originally exists in the group/device or not) **if some JEXL expression were translated**.
+* `jexlall`: changes expression from legacy to JEXL equivalence in the [fields where expressions may be used](#fields-in-which-expressions-may-be-used) (based on the translation dictionary specified by `--translations`). In addition, it set `expressionLanguage` field to `jexl` (no matter if the field originally exists in the group/device or not) **no matter if some JEXL expression were translated or not**.
```
Comment (mapedraza) [inline scripts/legacy_expression_tool/README.md:112]:
```suggestion
* `jexlall`: changes expression from legacy to JEXL equivalence in the [fields where expressions may be used](#fields-in-which-expressions-may-be-used) (based on the translation dictionary specified by `--translations`). In addition, it set `expressionLanguage` field to `jexl` (no matter if the field originally exists in the group/device or not). The difference between `jexl`and `jexlall` is in the second case, the script is not only looking for documents that contains legacy expressions, it also includes all groups that have `expressionLanguage` field defined. The `expressionLanguage` field on those documents (and also on the documents that contains legacy expresions) is set to `jexl`.
```

File: scripts/legacy_expression_tool/README.md [hunk 6]
```diff
@@ -104,6 +104,15 @@ The list of possible arguments that the scripts accepts are:
...
+* `jexlall`: changes expression from legacy to JEXL equivalence in the [fields where expressions may be used](#fields-in-which-expressions-may-be-used) (based on the translation dictionary specified by `--translations`). In addition, it set `expressionLanguage` field to `jexl` (no matter if the field originally exists in the group/device or not). The difference between `jexl`and `jexlall` is in the second case, the script is not only looking for documents that contains legacy expressions, it also includes all groups/devices that have `expressionLanguage` field defined. The `expressionLanguage` field on those documents (and also on the documents that contains legacy expresions) is set to `jexl`.
+
+The different between `jexl` and `jexall` is that in the former case the `expressionLanuage` field is set to `jexl` only if some expression were translated and the latter sets the field always, no matter if some expression were translated or not.
+
```
Comment (fgalan) [inline scripts/legacy_expression_tool/README.md:115]:
```suggestion
```

**Review by mapedraza — APPROVED** [review]
LGTM
